src/hooks/v2/V2ContractLoader.ts

```diff
@@ -46,11 +65,16 @@ export function useV2ContractLoader() {
   return contracts
 }
 
-const loadContract = (
-  contractName: keyof typeof V2ContractName,
+const loadContract = async (
+  contractName: V2ContractName,
   network: NetworkName,
   signerOrProvider: JsonRpcSigner | JsonRpcProvider,
-): Contract | undefined => {
-  const contract = require(`@jbx-protocol/contracts-v2/deployments/${network}/${contractName}.json`)
+): Promise<Contract | undefined> => {
+  const resolvedContractName =
+    CONTRACT_ABI_OVERRIDES[contractName] ?? contractName
+
+  const contract = await import(
```

Comment:
This is to avoid adding 1.5MB to a single file in the bundle

https://webpack.js.org/guides/code-splitting/#dynamic-imports